src/lib/moment/calendar.js

```diff
@@ -7,13 +7,23 @@ export function calendar (time, formats) {
     // Getting start-of-today depends on whether we're local/utc/offset or not.
     var now = time || createLocal(),
         sod = cloneWithOffset(now, this).startOf('day'),
-        diff = this.diff(sod, 'days', true),
-        format = diff < -6 ? 'sameElse' :
-            diff < -1 ? 'lastWeek' :
-            diff < 0 ? 'lastDay' :
-            diff < 1 ? 'sameDay' :
-            diff < 2 ? 'nextDay' :
-            diff < 7 ? 'nextWeek' : 'sameElse';
+        daysDiff = this.diff(sod, 'days', true),
+        monthsDiff = this.month() - sod.month(),
+        yearsDiff = this.year() - sod.year(),
+        format = daysDiff < -6 ?
+                    (yearsDiff === 0 ?
+                        (monthsDiff === 0 ? 'sameMonth' :
+                            monthsDiff === -1 ? 'lastMonth' : 'sameYear') :
+                    yearsDiff === -1 ? 'lastYear' : 'sameElse') :
```

Comment:
This expression is a little hard for me to understand. I think it can be re-ordered for the sake of readability. Wouldn't it be the case that you could start with the following:
```js
yearsDiff < -1 ? sameElse : 
(yearsDiff === -1 && monthsDiff < -1) ? lastYear : 
(monthsDiff === -1 && daysDiff < -6) ? lastMonth : 
```
from here I think the code can continue as written.